### PR TITLE
Adds `Tasks, `TaskList`, `TaskManager`.

### PR DESCRIPTION
**Pull request by junnbang**

### CODE REVIEW

File: src/main/java/seedu/address/model/task/Task.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Task task = (Task) o;
+        return getId() == task.getId();
```
Comment (tysng) [inline src/main/java/seedu/address/model/task/Task.java]: Add a special note here. Maybe tag it as `HACK` or something.. May trip on this in the future.

File: src/main/java/seedu/address/model/task/TaskManager.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class TaskManager {
+
+    private TaskList tasks;
+
+    public TaskManager() {
+        tasks = new TaskList();
+    }
+
+    public TaskManager(TaskList savedTasks) {
+        this();
```
Comment (tysng) [inline src/main/java/seedu/address/model/task/TaskManager.java]: Same as `tasks = savedTasks`?

File: src/main/java/seedu/address/model/task/TaskList.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+ * of list operations. List sort and filter operations are provided.
+ */
+public class TaskList {
+
+    private static int lastTaskId;
+    private final ObservableList<Task> tasks = FXCollections.observableArrayList();
+    private final ObservableList<Task> tasksUnmodifiable = FXCollections.unmodifiableObservableList(tasks);
+
+    public TaskList() {
+        lastTaskId = 0;
```
Comment (tysng) [inline src/main/java/seedu/address/model/task/TaskList.java]: Static variable's values should not be initiated in class constructor.. You will never know who will call another `TaskList()` somewhere and just throw it away.. Init in the attribute field.

**Review by tysg — CHANGES_REQUESTED** [review]

**Comment by junnbang** [comment]
All the changes are made.

**Review by tysg — APPROVED** [review]
LGTM

**Pull request merged**